lbz/configuration.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABCMeta, abstractmethod
+from os import getenv
+from typing import Any, Callable, Generic, Optional, TypeVar
+
+from lbz.aws_ssm import SSM
+from lbz.exceptions import ConfigurationMissingError, ConfigurationParsingError
+
+T = TypeVar("T")
+
+
+class ConfigValue(Generic[T], metaclass=ABCMeta):
+    """Mandatory Configuration
+
+    This class is not supporting None as outcome value.
+    """
+
+    def __init__(
+        self,
+        key: str,
+        parser: Callable[[str], T] = str,  # type: ignore
+        default: T = None,
+    ):
+        self._key = key
+        self._parser = parser
+        self._default = default
+        self._value: Optional[T] = None
+
+    @abstractmethod
+    def getter(self) -> Any:
+        pass
+
+    @property
+    def value(self) -> T:
+        if self._value is None:
+            val = self.getter()
+            if val is not None:
+                self._value = self._parse_value(val)
+            elif self._default is not None:
+                self._value = self._default
+            else:
+                raise ConfigurationMissingError(f"'{self._key}' was not defined.")
+
+        return self._value
+
+    def _parse_value(self, value: Any) -> T:
+        try:
+            return self._parser(value)
+        except Exception as error:
+            message = f"'{self._key}' could not be parsed with '{self._parser}'"
```

Comment:
This will be moved directly to the Exception but I think much more valuable is to put here the value that we tried to parse, instead of the representation of the parser function 😉